enhancements/node-tuning/pao-in-nto.md

```diff
@@ -0,0 +1,280 @@
+
+### Bugs
+
+PAO bugs are currently tracked in The OpenShift Bugzilla component [Performance Addon Operator](https://bugzilla.redhat.com/buglist.cgi?component=Performance%20Addon%20Operator&list_id=12053428&product=OpenShift%20Container%20Platform)
+
+This should be kept as-is, the Telco team will own all the bugs. We might need to move the component under NTO or rename it.
+
+The bugzilla component is currently not part of OCP metrics and we can include it. TBD
+
+
+### User Stories
+
+Both the NTO and PAO functionality should be kept intact and no regressions are allowed. The user should not notice anything except the fact that OLM is no longer needed to enable the tuning functionality.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions [optional]
+
+### Upgrade
+
+- PAO has been installed via OLM since 4.4. Once it moves to core we will have to figure out the upgrade procedure, because OLM will not remove the optional operator by itself.
+
+  One option is to remove the relevant OLM artifacts (CSV,Subscription,Operator group) via the performance-addon controller and leave the PAO CRDs without the change.
+
+  Another option, in case leader election supports such conditioning, is to keep using the same leadership election method and make sure the newest version always wins. The OLM operator will then stay on the cluster (can be release noted) as an inactive component.
+
+
+### Test Plan
+
+All tests already exist in OCP CI, the trigger will move to the NTO repository.
+
+Continuous testing is done via [OpenShift CI jobs](https://github.com/openshift/release/tree/daaab71832999c072c60667955637fa3a535d4ba/ci-operator/config/openshift-kni/performance-addon-operators) as well as some downstream specific Jenkins instances (both virtual and bare metal).
+
+### Graduation Criteria
+
+The user facing feature is already considered stable and GA, just shipped as an optional operator.
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+It is expected that the image will be kept up-to-date as part of regular OCP flows.
+
+API changes are handled by a migration webhook that PAO already contains to make upgrades transparent to users.
```

Comment:
what does this mean? What does it do?

Reply:
Conversion webhook, it makes sure that CRs created with older apiVersion are treated properly keeping the old behavior intact or using proper defaults.

Reply:
Do you have plans to deprecate v1 and eventually remove it? Or do you plan to roundtrip through v1 forever?

Reply:
Probably yes, we introduced v2 in 4.7 and there were no upgrade incompatible changes to it since. Deprecating v1 is something we can consider indeed, we just haven't had the time to do so yet.

Reply:
@MarSik  should we consider v1 deprecation as a design goal here ?

Reply:
Let's either do the v1 deprecation work before moving or after, but not at the same time.

Reply:
v1 deprecation work will happen after moving to NTO. 
In the meanwhile https://github.com/openshift-kni/performance-addon-operators/pull/727 should cover the back and forth conventions between v1 and v2 as mentioned above

Reply:
s/migration webhook/conversion webhook/